src/components/MapContainer/Layers/VectorLayer.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useContext, useEffect } from 'react'
+import MapContext from '../Map/MapContext'
+import OLVectorLayer from 'ol/layer/Vector'
+import { vector } from '../Source'
+import GeoJSON from 'ol/format/GeoJSON'
+import { get } from 'ol/proj'
+
+// note: both the VectorLayer styleOptions object and the 'source' from line 33 will need to be hoisted to be able to make multiple different vector layers for different data sources.
```

Comment:
this line's too long to read on github 😆  I'm an old school vimmer, 80 chars is ideal, although I know a lot of people are fine with 120 these days!  up to you.